### PR TITLE
IPv6 Support

### PR DESCRIPTION
**Pull request by briantopping**
Add configuration support for enabling IPv6.

### CODE REVIEW

**Comment by gardener-robot-ci-1** [comment]
Thank you @briantopping for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

File: Makefile [hunk 1]
```diff
@@ -14,8 +14,8 @@
 
 EXTENSION_PREFIX            := gardener-extension
 NAME                        := networking-cilium
-REGISTRY                    := eu.gcr.io/gardener-project
```
Comment (DockToFuture) [inline Makefile:17]: Please keep the gardener registry and image prefix. I will try out the changes later. Thx for the PR.

Comment (briantopping) [reply Makefile]: Argh, that leaked in, thank you for catching that. Fixed and rebased.

**Review by DockToFuture — CHANGES_REQUESTED** [review]
Just a small thing. :)

File: pkg/charts/utils.go [hunk 2]
```diff
@@ -154,5 +154,10 @@ func generateChartValues(config *ciliumv1alpha1.NetworkConfig, network *extensio
 		}
 	}
 
+	// check if IPv6 is enabled
+	if config.IPv6 != nil {
+		globalConfig.Ipv6.Enabled = *config.IPv6
```
[the remark this answers is not in the document]
Comment (briantopping) [reply pkg/charts/utils.go:159]: Thank you for the review. I saw both patterns. Could I learn from you how this is better? 🤔🤔

[the remark this answers is not in the document]
Comment (DockToFuture) [reply pkg/charts/utils.go:159]: Yes, there are both patterns and we were not consistent in the beginning. It's more ore less time for some spring cleaning.
In my eyes 
```
ipv6:
   enabled: true
```
is better readable than
```
ipv6: true
```
. But his are just nits. Thank you anyway for the PR. :)

File: pkg/apis/cilium/types_network.go [hunk 3]
```diff
@@ -88,6 +88,12 @@ type Hubble struct {
 	Enabled bool
 }
 
+// IPv6 enablement for cilium
+type IPv6 struct {
+	// Enabled indicates whether IPv6 is enabled or not.
+	Enabled bool `json:"enabled"`
```
Comment (DockToFuture) [inline pkg/apis/cilium/types_network.go:94]:
```suggestion
	Enabled bool
```

Comment (briantopping) [reply pkg/apis/cilium/types_network.go]: My bad, thanks for catching that.

**Comment by briantopping** [comment]
Changes resolved and rebased to branch. Thank you!

**Review by DockToFuture — APPROVED** [review]
/lgtm

**Comment by briantopping** [comment]
Good morning @DockToFuture, does this test seem stuck? Where would I look for it?

**Pull request merged**